worksheet_B/Mandelbrot.cpp

```diff
@@ -3,6 +3,125 @@
 
 #include "stdafx.h"
 
+
+//hsv to rgb converter https://www.programmingalgorithms.com/algorithm/hsv-to-rgb?lang=C%2B%2B
+class RGBvalue
+{
+public:
+	unsigned char R;
+	unsigned char G;
+	unsigned char B;
+
+	RGBvalue(unsigned char r, unsigned char g, unsigned char b)
+	{
+		R = r;
+		G = g;
+		B = b;
+	}
+
+	bool Equals(RGBvalue rgb)
+	{
+		return (R == rgb.R) && (G == rgb.G) && (B == rgb.B);
+	}
+};
+
+class HSV
+{
+public:
+	double H;
+	double S;
+	double V;
+
+	HSV(double h, double s, double v)
+	{
+		H = h;
+		S = s;
+		V = v;
+	}
+
+	bool Equals(HSV hsv)
+	{
+		return (H == hsv.H) && (S == hsv.S) && (V == hsv.V);
+	}
+};
+
+static RGBvalue HSVToRGBconverter(HSV hsv) {
+	double r = 0, g = 0, b = 0;
+
+	if (hsv.S == 0)
+	{
+		r = hsv.V;
+		g = hsv.V;
+		b = hsv.V;
+	}
+	else
+	{
+		int i;
+		double f, p, q, t;
+
+		if (hsv.H == 360)
+			hsv.H = 0;
+		else
+			hsv.H = hsv.H / 60;
+
+		i = (int)trunc(hsv.H);
+		f = hsv.H - i;
+
+		p = hsv.V * (1.0 - hsv.S);
+		q = hsv.V * (1.0 - (hsv.S * f));
+		t = hsv.V * (1.0 - (hsv.S * (1.0 - f)));
+
+		switch (i)
+		{
+		case 0:
+			r = hsv.V;
+			g = t;
+			b = p;
+			break;
+
+		case 1:
+			r = q;
+			g = hsv.V;
+			b = p;
+			break;
+
+		case 2:
+			r = p;
+			g = hsv.V;
+			b = t;
+			break;
+
+		case 3:
+			r = p;
+			g = q;
+			b = hsv.V;
+			break;
+
+		case 4:
+			r = t;
+			g = p;
+			b = hsv.V;
+			break;
+
+		default:
+			r = hsv.V;
+			g = p;
+			b = q;
+			break;
+		}
+
+		if (hsv.H > 200)
```

Comment:
This looks like a program-specific change to the standard HSV-to-RGB algorithm -- probably better to move this out into the main function. (So that next time you need to convert HSV to RGB, and you decide to reuse this code, you don't find yourself wondering why it returns the wrong value for certain inputs...)